src/locks.rs

```diff
@@ -37,7 +39,8 @@ macro_rules! recover_lock {
 /// This macro unlocks a Mutex or RwLock. If the lock is poisoned (i.e. a panic before a MutexGuard / RwLockGuard is
 /// dropped) the last value before the panic occurred is used.
 ///
-/// This macro should not be used if the implementation should fail a if the lock was poisoned.
+/// This macro should not be used if the implementation should fail or if the lock was poisoned.
+/// !!! Above typo fix should be double checked !!!
```

Comment:
@CjS77 pls confirm the above typo fix.

Reply:
I think it should be "This macro should not be used if the implementation should fail when the lock was poisoned."
@sdbondi can confirm

Reply:
The semantics are akin to rollback on a database transaction (See `Mutex::lock`). Might be best to frame it that way so that any user of the macro can decide if those semantics are appropriate. 

```rust
/// This macro returns a Mutex or RwLock guard without returning a `PoisonError`. If the lock is poisoned (i.e. a panic before a MutexGuard / RwLockGuard is dropped) 
/// the last value before the panic occurred is used.
// The semantics of this macro are similar to a database transaction rollback on failure.
```

Reply:
Updated per @sdbondi 's comment above.